swift/ios-shoppe-demo/Views/StoreViewController.swift

```diff
@@ -70,16 +77,32 @@ class StoreViewController: UICollectionViewController, UICollectionViewDelegateF
         let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as? ProductCollectionViewCell
         cell?.product = products[indexPath.row]
         cell?.collectionView = self
-        
+
         return cell ?? UICollectionViewCell()
     }
 
     func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
-        
+
         return CGSize(width: 375, height: 300)
     }
 
     func addToCart(_ product: String) {
+        let generator = UIImpactFeedbackGenerator(style: .medium)
+        generator.impactOccurred()
+
         order.addProduct(product)
+        updateCartNumber()
+        setupNavigationBar()
+    }
+
+    func updateCartNumber() {
+        let orderItems = order.items.filter({ $0.quantity > 0 })
```

Comment:
You could try out another reduce here.  Rather than iterate over the list to filter items and then iterate again to count, you could simply assign cartNumber to the value returned from reduce. The reduce should iterate over the list and check if item is > 0 and if so, add it and return it to next.